cdap-app-fabric/src/test/java/co/cask/cdap/app/runtime/monitor/RuntimeMonitorTest.java

```diff
@@ -21,195 +21,149 @@
-import com.google.gson.Gson;
 import com.google.inject.Guice;
 import com.google.inject.Injector;
-import com.sun.net.httpserver.HttpExchange;
-import com.sun.net.httpserver.HttpHandler;
-import com.sun.net.httpserver.HttpServer;
 import org.junit.After;
 import org.junit.Before;
-import org.junit.Ignore;
 import org.junit.Test;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import java.io.IOException;
-import java.net.HttpURLConnection;
 import java.net.InetAddress;
-import java.net.InetSocketAddress;
-import java.util.ArrayList;
-import java.util.HashMap;
-import java.util.HashSet;
-import java.util.LinkedHashMap;
-import java.util.LinkedHashSet;
-import java.util.List;
-import java.util.Map;
+import java.util.UUID;
 import java.util.concurrent.Callable;
 import java.util.concurrent.TimeUnit;
 
 /**
  * Runtime Monitor Test
  */
-@Ignore
 public class RuntimeMonitorTest {
   private static final Logger LOG = LoggerFactory.getLogger(RuntimeMonitorTest.class);
 
   protected static Injector injector;
   protected static CConfiguration cConf;
   protected static MessagingService messagingService;
-  private static final Gson GSON = new Gson();
-  private HttpServer httpServer;
-  private InetSocketAddress address;
+  private InetAddress address;
+  private RuntimeServer runtimeServer;
+  private MessagingContext messagingContext;
 
   @Before
-  public void init() throws IOException, UnsupportedTypeException {
+  public void init() throws IOException, UnsupportedTypeException, TopicAlreadyExistsException {
     cConf = CConfiguration.create();
     cConf.set(Constants.AppFabric.OUTPUT_DIR, System.getProperty("java.io.tmpdir"));
     cConf.set(Constants.AppFabric.TEMP_DIR, System.getProperty("java.io.tmpdir"));
-    cConf.set(Constants.AppFabric.PROGRAM_STATUS_EVENT_TOPIC, "cdap-status");
+    cConf.set(Constants.AppFabric.PROGRAM_STATUS_EVENT_TOPIC, "programStatus");
+    cConf.set(Constants.RuntimeHandler.SERVER_PORT, "1234");
     cConf.set(Constants.RuntimeMonitor.BATCH_LIMIT, "2");
-    cConf.set(Constants.RuntimeMonitor.POLL_TIME_MS, "20");
-    cConf.set(Constants.RuntimeMonitor.TOPICS, Constants.AppFabric.PROGRAM_STATUS_RECORD_EVENT_TOPIC);
+    cConf.set(Constants.RuntimeMonitor.POLL_TIME_MS, "200");
+    cConf.set(Constants.RuntimeMonitor.TOPICS_CONFIGS, Constants.AppFabric.PROGRAM_STATUS_EVENT_TOPIC);
     injector = Guice.createInjector(new AppFabricTestModule(cConf));
     messagingService = injector.getInstance(MessagingService.class);
     if (messagingService instanceof Service) {
       ((Service) messagingService).startAndWait();
     }
-    address = new InetSocketAddress(InetAddress.getLoopbackAddress(), 1234);
-    httpServer = HttpServer.create(address, 0);
-    httpServer.start();
+    address = InetAddress.getLoopbackAddress();
+    messagingContext = new MultiThreadMessagingContext(messagingService);
+    messagingService.createTopic(new TopicMetadata(new TopicId("system", "cdap-programStatus")));
```

Comment:
You shouldn't need to call this if you setup the cdap-default.xml correctly. It should be in the `messaging.system.topics` list. The topic will be created automatically when the messaging service starts.

Reply:
This is done to test the case where if a runtime started with latest cdap-site.xml. In that case`runtime` has different topic for program states than cdap runtime monitor.